src/libmongoc/src/mongoc/mongoc-cursor.c

```diff
@@ -790,16 +790,14 @@ _mongoc_cursor_monitor_succeeded (mongoc_cursor_t *cursor,
    bson_init (&docs_array);
    _mongoc_cursor_append_docs_array (cursor, &docs_array, response);
 
-   bson_init (&reply);
-   bson_append_int32 (&reply, "ok", 2, 1);
-   bson_append_document_begin (&reply, "cursor", 6, &reply_cursor);
-   bson_append_int64 (&reply_cursor, "id", 2, mongoc_cursor_get_id (cursor));
-   bson_append_utf8 (&reply_cursor, "ns", 2, cursor->ns, cursor->nslen);
-   bson_append_array (&reply_cursor,
-                      first_batch ? "firstBatch" : "nextBatch",
-                      first_batch ? 10 : 9,
-                      &docs_array);
-   bson_append_document_end (&reply, &reply_cursor);
+   bsonBuildDecl (reply,
+                  kv ("ok", int32 (1)),
+                  kv ("cursor",
+                      doc (kv ("id", int64 (mongoc_cursor_get_id (cursor))),
+                           kv ("ns", utf8_w_len (cursor->ns, cursor->nslen)),
+                           kv (first_batch ? "firstBatch" : "nextBatch",
+                               bsonArray (docs_array)))));
```

Comment:
```suggestion
   bsonBuildDecl (
      reply,
      kvl ("ok", 2, int32 (1)),
      kvl ("cursor",
           6,
           doc (kvl ("id", 2, int64 (mongoc_cursor_get_id (cursor))),
                kvl ("ns", 2, utf8_w_len (cursor->ns, cursor->nslen)),
                kvl (first_batch ? "firstBatch" : "nextBatch",
                     first_batch ? 10 : 9,
                     bsonArray (docs_array)))));
```
Suggest preserving lengths to avoid calls to `strlen`.

Reply:
The calls to strlen() will be (_should be_) constant-folded (by any competent optimizer) because this macro-expands to a call to strlen() of a string literal.